pkg/api/latest/dynakube/extension/spec.go

```diff
@@ -0,0 +1,144 @@
+package extension
+
+import (
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/shared/image"
+	"github.com/Dynatrace/dynatrace-operator/pkg/consts"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type Extensions struct {
+	ExecutionController    *ExecutionControllerSpec
+	OpenTelemetryCollector *OpenTelemetryCollectorSpec
+
+	name      string
+	namespace string
+	enabled   bool
+}
+
+func (e *Extensions) SetName(name string) {
+	e.name = name
+}
+
+func (e *Extensions) SetNamespace(namespace string) {
+	e.namespace = namespace
+}
+
+func (e *Extensions) SetEnabled(enabled bool) {
+	e.enabled = enabled
+}
+
+func (e *Extensions) Enabled() bool {
```

Comment:
```suggestion
func (e *Extensions) IsEnabled() bool {
```

to follow the pattern in other packages